### PR TITLE
at-cb: Support Atemis sensor reading

### PR DESCRIPTION
**Pull request by DelphineChiu**
Summary:
- Support tmp75 sensor reading.
- Support ADC sensor reading.
- Support xdpe15284 sensor reading.
- Support ADM1272 sensor reading.
- Support INA233 sensor reading.
- Support PEX89144 sensor reading.
- Support Q50SN120A1 sensor reading.

Test Plan:
- Build code: Pass

### CODE REVIEW

File: meta-facebook/at-cb/src/platform/plat_class.c [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+bool get_adc_voltage(int channel, float *voltage)
+{
+	CHECK_NULL_ARG_WITH_RETURN(voltage, false)
+
+	if (channel >= NUMBER_OF_ADC_CHANNEL) {
+		LOG_ERR("Invalid ADC channel-%d", channel);
+		return false;
+	}
+
+	uint32_t raw_value = 0, reg_value = 0;
```
Comment (GoldenBug) [inline meta-facebook/at-cb/src/platform/plat_class.c:59]: Can we make this 2 separate lines?

```
uint32_t raw_value = 0;
uint32_t reg_value = 0;
```

File: meta-facebook/at-cb/src/platform/plat_hook.c [hunk 2]
```diff
@@ -15,15 +15,139 @@
...
+	// Select Channel
+	bool ret = true;
+	int mutex_status = 0;
+	mux_config *pre_args = (mux_config *)args;
+	pre_args->bus = sensor_config[sensor_config_index_map[sensor_num]].port;
+
+	struct k_mutex *mutex = get_i2c_mux_mutex(pre_args->bus);
+	mutex_status = k_mutex_lock(mutex, K_MSEC(MUTEX_LOCK_INTERVAL_MS));
+	if (mutex_status != 0) {
+		LOG_ERR("pre_ina233_read: mutex lock fail, status: %d", mutex_status);
```
Comment (GoldenBug) [inline meta-facebook/at-cb/src/platform/plat_hook.c:85]: It should already print the function name as part of the log automatically.
So "pre_ina233_read:" isn't needed in the message.

File: meta-facebook/at-cb/src/platform/plat_hook.c [hunk 3]
```diff
@@ -15,15 +15,139 @@
...
+	int unlock_status = 0;
+	uint8_t bus = sensor_config[sensor_config_index_map[sensor_num]].port;
+
+	struct k_mutex *mutex = get_i2c_mux_mutex(bus);
+	if (mutex->lock_count != 0) {
+		unlock_status = k_mutex_unlock(mutex);
+	}
+
+	if (unlock_status != 0) {
+		LOG_ERR("post_ina233_read: unlock fail, status: %d", unlock_status);
```
Comment (GoldenBug) [inline meta-facebook/at-cb/src/platform/plat_hook.c:107]: Same, can remove function name from beggining.

File: meta-facebook/at-cb/src/platform/plat_hook.c [hunk 4]
```diff
@@ -15,15 +15,139 @@
...
+
+	bool ret = true;
+	int mutex_status = 0;
+	mux_config *pre_args = (mux_config *)args;
+	pre_args->bus = sensor_config[sensor_config_index_map[sensor_num]].port;
+
+	struct k_mutex *mutex = get_i2c_mux_mutex(pre_args->bus);
+	mutex_status = k_mutex_lock(mutex, K_MSEC(MUTEX_LOCK_INTERVAL_MS));
+	if (mutex_status != 0) {
+		LOG_ERR("pre_pex_read: mutex lock fail, status: %d", mutex_status);
```
Comment (GoldenBug) [inline meta-facebook/at-cb/src/platform/plat_hook.c:126]: Remove pre_pex_read

File: meta-facebook/at-cb/src/platform/plat_hook.c [hunk 5]
```diff
@@ -15,15 +15,139 @@
...
+	int unlock_status = 0;
+	uint8_t bus = sensor_config[sensor_config_index_map[sensor_num]].port;
+
+	struct k_mutex *mutex = get_i2c_mux_mutex(bus);
+	if (mutex->lock_count != 0) {
+		unlock_status = k_mutex_unlock(mutex);
+	}
+
+	if (unlock_status != 0) {
+		LOG_ERR("post_pex_read: unlock fail, status: %d", unlock_status);
```
Comment (GoldenBug) [inline meta-facebook/at-cb/src/platform/plat_hook.c:148]: remove post_pex_read

File: meta-facebook/at-cb/src/platform/plat_sensor_table.c [hunk 6]
```diff
@@ -54,3 +320,51 @@ void load_sensor_config(void)
...
+		mutex = &i2c_2_tca9543_mutex;
+		break;
+	case I2C_BUS3:
+		mutex = &i2c_3_tca9543_mutex;
+		break;
+	case I2C_BUS4:
+		mutex = &i2c_4_pi4msd5v9542_mutex;
+		break;
+	default:
+		LOG_ERR("get_mutex_func: not support i2c bus %d mutex\n", i2c_bus);
```
Comment (GoldenBug) [inline meta-facebook/at-cb/src/platform/plat_sensor_table.c:339]:
```
LOG_ERR("No support for i2c bus %d mutex", i2c_bus);
```

File: meta-facebook/at-cb/src/platform/plat_sensor_table.c [hunk 7]
```diff
@@ -54,3 +320,51 @@ void load_sensor_config(void)
...
+	float mb_dc_low_threshold = P1V8_PEX_LOW_THRESHOLD;
+	float voltage = 0;
+
+	ret = get_adc_voltage(channel, &voltage);
+	if (ret == true) {
+		if (voltage < mb_dc_low_threshold) {
+			ret = false;
+		}
+	} else {
+		LOG_ERR("mb_dc_on func: ret false\n");
```
Comment (GoldenBug) [inline meta-facebook/at-cb/src/platform/plat_sensor_table.c:361]: Remove function name.
Remove newline.

**Review by GoldenBug — CHANGES_REQUESTED** [review]

File: meta-facebook/at-cb/src/platform/plat_sdr_table.c [hunk 8]
```diff
@@ -24,7 +24,3909 @@
 #include "plat_sensor_table.h"
 
 SDR_Full_sensor plat_sdr_table[] = {
```
Comment (GoldenBug) [inline meta-facebook/at-cb/src/platform/plat_sdr_table.c:26]: GT isn't planned to use IPMI correct?
Is the SDR table necessary then?

**Review by GoldenBug — CHANGES_REQUESTED** [review]

File: meta-facebook/at-cb/src/platform/plat_sdr_table.c (hunk 8 above)
Comment (LoraLin1) [reply meta-facebook/at-cb/src/platform/plat_sdr_table.c:26]: Hi @GoldenBug ,
BIC needs to use IPMI to communicate with BMC, but the protocol between BIC and BMC doesn't use IPMB.
BIC needs to record the SDR table to provide the SDR of the sensor to the BMC.

**Pull request closed without merging**